power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -49,9 +49,7 @@ class MainModel {
     };
     ~MainModel() { impl_.reset(); }
 
-    static bool is_update_independent(ConstDataset const& update_data) {
-        return Impl::is_update_independent(update_data);
-    }
+    bool is_update_independent(ConstDataset const& update_data) { return impl().is_update_independent(update_data); }
```

Comment:
I thought we want to per component look if the update is independent or not. Now this is still one boolean for the whole dataset?

Reply:
Will be added.

Reply:
As discussed, will become part of a follow up.